Visualization: `plot_tabular` does not work for x as `np.ndarray`

The input `x` in `plot_tabular` is defined as `Array of feature importance scores`. However, it returns an error if `x` is `np.ndarray`. There isnot a test for this in` test_visualization.py`. You can reproduce the error as:

```python
import numpy as np
def test_plot_tabular_with_array():
    """Test plot tabular data."""
    x = np.random.rand(5, 3)
    y = [f"Feature {i}" for i in range(x.shape[1])]
    fig, ax = plot_tabular(x=x, y=y, show_plot=False)
    assert fig is not None
```

the error
```bash
>       top_values = [x for _, x in sorted(zip(abs_values, x), reverse=True)][:num_features]
E       ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

--- a/dianna/visualization/tabular.py
+++ b/dianna/visualization/tabular.py
@@ -18,7 +18,7 @@ def plot_tabular(
     """Plot feature importance with segments highlighted.
 
     Args:
-        x (np.ndarray): Array of feature importance scores
+        x (np.ndarray): 1D array of feature importance scores of one instance
         y (List[str]): List of feature names
         x_label (str): Label for the x-axis
         y_label (str): Label or list of labels for the y-axis
@@ -32,6 +32,12 @@ def plot_tabular(
     Returns:
         plt.Figure
     """
+    # check type and shape of x should be 1D array
+    if not isinstance(x, np.ndarray):
+        raise TypeError("x should be a numpy array")
+    if x.ndim != 1:
+        raise ValueError("x should be a 1D array")
+
     if not num_features:
         num_features = len(x)
     abs_values = [abs(i) for i in x]

--- a/tests/test_visualization.py
+++ b/tests/test_visualization.py
@@ -19,6 +19,7 @@ def test_plot_tabular(tmpdir):
     assert output_path.exists()
 
 def test_plot_tabular_with_ndarray():
+    """Test plot tabular data with ndarray."""
     x = np.random.rand(5, 3)
     y = [f"Feature {i}" for i in range(x.shape[1])]
     # check ValueError

--- a/tests/test_visualization.py
+++ b/tests/test_visualization.py
@@ -18,6 +18,13 @@ def test_plot_tabular(tmpdir):
 
     assert output_path.exists()
 
+def test_plot_tabular_with_ndarray():
+    x = np.random.rand(5, 3)
+    y = [f"Feature {i}" for i in range(x.shape[1])]
+    # check ValueError
+    with pytest.raises(ValueError):
+        plot_tabular(x=x, y=y, show_plot=False)
+
 
 def test_plot_timeseries_univariate(tmpdir, random):
     """Test plot univariate time series."""

